### PR TITLE
Order resources by their most recent update state version

### PR DESCRIPTION
**Pull request by KrzysztofLabus-RDX**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/EntityResourceAggregateHistory.cs [hunk 1]
```diff
@@ -84,12 +86,41 @@ internal class EntityResourceAggregateHistory
...
+    [Column("non_fungible_resource_significant_update_state_versions")]
+    public List<long> NonFungibleResourceSignificantUpdateStateVersions { get; set; }
+
+    public void MarkAsCloneOfExistingRecord()
+    {
+        _fungibleResourceEntityIdsOrg = FungibleResourceEntityIds.ToArray();
+        _nonFungibleResourceEntityIdsOrg = NonFungibleResourceEntityIds.ToArray();
+    }
+
+    public bool ShouldBePersisted()
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Models/EntityResourceAggregateHistory.cs:104]: that piece of code would be easily testable by unit tests, maybe we should add some?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/EntityResourceAggregateHistory.cs [hunk 2]
```diff
@@ -84,12 +86,41 @@ internal class EntityResourceAggregateHistory
     [Column("fungible_resource_entity_ids")]
     public List<long> FungibleResourceEntityIds { get; set; }
 
-    [Column("fungible_resource_last_update_state_versions")]
-    public List<long> FungibleResourceLastUpdateStateVersions { get; set; }
+    [Column("fungible_resource_significant_update_state_versions")]
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Models/EntityResourceAggregateHistory.cs:89]: I could come with better name for these columns so I think we have to leave them as it is. But if we can't find better name, what do you think about adding comment here to explaining what it is about?

**Review by PawelPawelec-RDX — APPROVED** [review]
just minor comments, feel free to merge.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/EntityResourceAggregateHistory.cs (hunk 1 above)
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Models/EntityResourceAggregateHistory.cs:192]: Refactored EntityResourceAggregateHistory quite heavily.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/EntityResourceAggregateHistory.cs (hunk 2 above)
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Models/EntityResourceAggregateHistory.cs:99]: Refactored EntityResourceAggregateHistory quite heavily.

**Review by PawelPawelec-RDX — APPROVED** [review]

**Pull request merged**